File: melant_ia/src/modules/02_asistente_tecnico_rural/asistente_logic.py
import json
import os

def calcular_fertilizacion(cultivo_nombre, toneladas_esperadas):
    # 1. Cargar los datos offline
    ruta_json = os.path.join(os.path.dirname(__file__), "data_cultivos.json")
    
    with open(ruta_json, 'r', encoding='utf-8') as f:
        datos = json.load(f)
    
    if cultivo_nombre not in datos["requerimientos"]:
        return "Cultivo no encontrado en la base de datos offline."

    req = datos["requerimientos"][cultivo_nombre]
    
    # 2. Lógica de cálculo (Extracción nutricional)
    n_total = req["N"] * toneladas_esperadas
    p_total = req["P"] * toneladas_esperadas
    k_total = req["K"] * toneladas_esperadas

    # 3. Formatear resultado para el usuario
    resultado = f"""
    📊 PLAN DE FERTILIZACIÓN (Estimado)
    ----------------------------------
    Cultivo: {req['nombre']}
    Meta de Producción: {toneladas_esperadas} Toneladas
    
    Necesidad total de Nutrientes:
    - Nitrógeno (N): {n_total:.2f} kg
    - Fósforo (P): {p_total:.2f} kg
    - Potasio (K): {k_total:.2f} kg
    
    Nota: Consulte con un técnico para ajustar según su análisis de suelo.
    """
    return resultado

# Prueba rápida
if __name__ == "__main__":
    print(calcular_fertilizacion("naranja", 10))

File: melant_ia/src/modules/02_asistente_tecnico_rural/asistente_logic.js
// asistente_logic.js
// Adaptación JS del módulo Python asistente_logic.py para integración web

export function calcularFertilizacion(cultivoNombre, toneladasEsperadas, data) {
  // data debe ser un objeto con la estructura de data_cultivos.json
  if (!data || !data.requerimientos || !data.requerimientos[cultivoNombre]) {
    return 'Cultivo no encontrado en la base de datos offline.';
  }
  const req = data.requerimientos[cultivoNombre];
  // Lógica de cálculo (Extracción nutricional)
  const nTotal = req.N * toneladasEsperadas;
  const pTotal = req.P * toneladasEsperadas;
  const kTotal = req.K * toneladasEsperadas;
  // Formatear resultado para el usuario
  return `📊 PLAN DE FERTILIZACIÓN (Estimado)\n----------------------------------\nCultivo: ${req.nombre}\nMeta de Producción: ${toneladasEsperadas} Toneladas\n\nNecesidad total de Nutrientes:\n- Nitrógeno (N): ${nTotal.toFixed(2)} kg\n- Fósforo (P): ${pTotal.toFixed(2)} kg\n- Potasio (K): ${kTotal.toFixed(2)} kg\n\nNota: Consulte con un técnico para ajustar según su análisis de suelo.`;
}

// Ejemplo de uso:
// import { calcularFertilizacion } from './asistente_logic.js';
// const resultado = calcularFertilizacion('naranja', 10, data_cultivos_json);
// console.log(resultado);
